les/api.go

```diff
@@ -352,3 +419,60 @@ func (api *PrivateLightAPI) GetCheckpointContractAddress() (string, error) {
 	}
 	return api.backend.oracle.config.Address.Hex(), nil
 }
+
+//API should be for light clients of les protocol
+type LightClientAPI struct {
+	le *LightEthereum
+}
+
+func NewLightClientAPI(le *LightEthereum) *LightClientAPI {
+	return &LightClientAPI{le}
+}
+
+// PullPeerGatewayFees updates cache by pulling gateway fee peer nodes
+func (api *LightClientAPI) GatewayFeeCache() map[string]*gatewayFeeEtherbase {
+	return api.le.handler.gatewayFeeCache.gatewayFeeMap
+}
+
+// PullPeerGatewayFees updates cache by pulling gateway fee peer nodes
+func (api *LightClientAPI) PullPeerGatewayFees() error {
+	peerNodes := api.le.peers.AllPeers()
+	if len(peerNodes) == 0 {
+		return errors.New("No peers")
+	}
+
+	reqId := genReqID()
+
+	for _, peerNode := range peerNodes {
+		cost := peerNode.GetRequestCost(GetGatewayFeeMsg, int(1))
+		err := peerNode.RequestGatewayFee(reqId, cost)
+		if err != nil {
+			continue
+		}
+	}
+	return nil
+}
+
+func (api *LightClientAPI) MinPeerGatewayFee() map[string]*gatewayFeeEtherbase { //get min gwFee as well as corresponding etherbase from cache. Can't return a tuple for some reason
```

Comment:
As we discussed, I would recommend renaming this to `SuggestGatewayFee` to allow for more generic implementations in the backend. This will be important for users of light clients that want to use a ledger or other signing mechanism outside of Geth